enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -0,0 +1,602 @@
+
+### Non-Goals
+
+- Configuring policies for the images in the OCP payload is not within the scope of this enhancement.
+- Providing a tool to mirror the signatures is out of the scope of this enhancement. In order to verify the signature, the disconnected users need to mirror signatures together with the application images. 
+- Grant the application administrator the ability to weaken cluster-scoped policies, to avoid expanding the set of administrators capable of increasing cluster exposure to vulnerable images.
+- Grant the application administrator the ability to tighten cluster-scoped policies. This could be useful in the future, but we are deferring it to limit the amount of work needed for an initial implementation.
+
+## Proposal
+
+### Workflow Description
+
+**cluster administrator** Signature Verification Configuration Workflow:
+1. The cluster administrator requests the addition of signature verification configurations at the cluster scope.
+2. The cluster administrator writes the verification certification to the ClusterImagePolicy YAML file and creates a ClusterImagePolicy CR using `oc create -f imagepolicy.yaml`.
+3. The cluster administrator can retrieve the merged cluster-scoped policies by checking `/etc/containers/policy.json` within the `99-<pool>-generated-registries` machine-config. 
+4. The cluster administrator has the option to delete the signature verification configuration by removing its ClusterImagePolicy instances. 
+
+**application administrator** Signature Verification Configuration Workflow:
+1. The application administrator requests the addition of signature verification configurations at the namespace scope.
+2. The application administrator writes the verification certification to the ImagePolicy YAML file and creates a ImagePolicy CR using `oc create -f imagepolicy.yaml`. Please note that the application administrator cannot override cluster-scoped policies, as they are treated with higher priority.
+3. The application administrator can retrieve the cluster override and merged policies by checking `<NAMESPACE>.json` within the `99-<pool>-generated-imagepolicies` machine-config. 
+4. The application administrator has the option to remove the signature verification configuration by deleting its ImagePolicy instances.
+
+### API Extensions
+
+#### Type definitions
+
+Type definitions of ImagePolicy. ClusterImagePolicy is expected to have a similar structure to ImagePolicy. 
+
+```go
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePolicy holds namespace-wide configuration for image signature verification
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+type ImagePolicy struct {
```

Comment:
A ClusterImagePolicy will look practically the same as this won't it?

Reply:
Yes, addressed that `ClusterImagePolicy is expected to have a similar structure to ImagePolicy.`